pkg/generate/app/cmd/newapp.go

```diff
@@ -269,8 +269,12 @@ func (c *AppConfig) validateBuilders(components app.ComponentReferences) error {
 }
 
 func validateEnforcedName(name string) error {
-	if reasons := validation.ValidateServiceName(name, false); len(reasons) != 0 && !app.IsParameterizableValue(name) {
-		return fmt.Errorf("invalid name: %s. Must be an a lower case alphanumeric (a-z, and 0-9) string with a maximum length of 24 characters, where the first character is a letter (a-z), and the '-' character is allowed anywhere except the first or last character.", name)
+	// The limiting factor on name length is that name may be substituted into
+	// imageapi.ContainerImageEntrypointAnnotationFormatKey, whose part after
+	// the '/' character cannot exceed 63 characters.
```

Comment:
74552466e64a1321191ad5862485dbbee751369e changed ValidateServiceName in vendor/k8s.io/kubernetes/pkg/api/validation/validation.go from NameIsDNS952Label to NameIsDNS1035Label, thus increasing the potential length from 24 to 63.

So, how do you want to proceed with this, given the myriad of potential possibilities?